Simulator/conversions.py

```diff
@@ -87,21 +90,33 @@ def NED_to_ECI(vec_NED, glat, glon, GMST):
 def L(q):
     s = q[0]
     v = q[1:4]
-    I3 = np.eye(3)
+    # I3 = np.eye(3)
 
-    upper = np.hstack([s, -v])
-    lower = np.column_stack([v, s*I3 + skew(v)])
-    return  np.row_stack([upper, lower])
+    # upper = np.hstack([s, -v])
+    # lower = np.column_stack([v, s*I3 + skew(v)])
+    Vhat = skew(v)
+    L = np.array([[s, -v[0], -v[1], -v[2]], 
+        [v[0], s, Vhat[0, 1], Vhat[0, 2]], 
+        [v[1], Vhat[1, 0], s, Vhat[1, 2]], 
+        [v[2], Vhat[2, 0], Vhat[2, 1], s]])
+    # return  np.row_stack([upper, lower])
+    return L
 

```

Comment:
In the spirit of avoiding any mistakes in the future, would you happen to know what the difference between these two implementations is? I went back to class notes and verified that the previous implementation matches the formulas.

Reply:
They are identical functionally, however, numpy is unreal slow. Just writing out the matrix is much faster than using any of the colstack, rowstack stuff. This is a speed thing.

Reply:
Phew! Glad there weren't bugs everywhere haha. Thanks for clarifying.